### PR TITLE
Ensure _closedTcs is set and fix issue where first request to complete is closing the channel

### PR DESCRIPTION
**Pull request by mconnew**
Fixes issue #1180

### CODE REVIEW

**Comment by mconnew** [comment]
@dotnet-bot test all outerloop please

**Comment by mconnew** [comment]
@dotnet-bot test all outerloop please

File: src/System.Private.ServiceModel/src/System/ServiceModel/Channels/RequestChannel.cs [hunk 1]
```diff
@@ -185,28 +187,36 @@ protected override void OnAbort()
...
+                    _outstandingRequests.Remove(request);
+                    var outstandingRequestCloseCount = Interlocked.Decrement(ref _outstandRequestCloseCount);
+
+                    if (outstandingRequestCloseCount == 0 && _closedTcs != null)
                     {
-                        // Protect against close altering _closedTcs concurrently
-                        var closedTcs = Interlocked.CompareExchange(ref _closedTcs, null, _closedTcs);
-                        if (closedTcs != null)
+                        // When we are closed or closing, _closedTcs is managed by the close logic.
+                        if (!_closed)
```
Comment (roncain) [inline src/System.Private.ServiceModel/src/System/ServiceModel/Channels/RequestChannel.cs]: Nit (not required to merge): I think you could combine this 'if (!_closed)' test with the if above.

**Comment by roncain** [comment]
LGTM.
Nit level suggestion can be ignored to facilitate faster checkin.
OuterLoop failures on *nix are expected and unrelated to this PR, so okay to merge from my standpoint.

**Pull request merged**